pkg/doc/verifiable/example_support_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable_test
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/piprate/json-gold/ld"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+)
+
+type UniversityDegree struct {
+	Type       string `json:"type,omitempty"`
+	University string `json:"university,omitempty"`
+}
+
+type UniversityDegreeSubject struct {
+	ID     string           `json:"id,omitempty"`
+	Name   string           `json:"name,omitempty"`
+	Spouse string           `json:"spouse,omitempty"`
+	Degree UniversityDegree `json:"degree,omitempty"`
+}
+
+type UniversityDegreeCredential struct {
+	*verifiable.Credential
+
+	ReferenceNumber int `json:"referenceNumber,omitempty"`
+}
+
+func (udc *UniversityDegreeCredential) MarshalJSON() ([]byte, error) {
+	// todo too complex! (https://github.com/hyperledger/aries-framework-go/issues/847)
+	c := udc.Credential
+	cp := *c
+
+	cp.CustomFields = map[string]interface{}{
+		"referenceNumber": udc.ReferenceNumber,
+	}
+
+	return json.Marshal(&cp)
+}
+
+func getEd25519Signer(privKey []byte) *ed25519Signer {
+	return &ed25519Signer{privateKey: privKey}
+}
+
+type ed25519Signer struct {
```

Comment:
is it necessary to create new signer definition in every file requiring ed25519 signer? can't we call a single default definition?